New flow for data deletion and editorial improvements (#629)
italia/eidas-it-wallet-docs: docs/en/relying-party-instance.rst

**Steps 9-11:** The Mobile Relying Party Instance:

  1. Generates an ``hardware_signature`` value by signing the ``client_data_hash`` with the Hardware Cryptographic private key, serving as a proof of possession for the Cryptographic Hardware Keys.
  2. Generates the :ref:`Relying Party Key Binding Request` in the form of a JWT. This JWT includes ``key_attestation``, ``hardware_signature``, ``nonce``, ``hardware_key_tag``, and ``cnf`` (representing ``key_pub``); it is signed using ``key_priv``.
  3. Sends the signed :ref:`Relying Party Key Binding Request` JWT as an ``assertion`` parameter in the body of an HTTP request to the :ref:`Relying Party Key Binding Endpoint`.

> **Step 12:** The :ref:`Relying Party Key Binding Endpoint` evaluates the Key Binding Request and performs the following checks:

Reviewer: ```suggestion
**Step 12:** The Relying Party Backend evaluates the Key Binding Request and performs the following checks:
```

  1. The request includes all required HTTP header parameters as defined in :ref:`Relying Party Key Binding Request`.
  2. The signature of the Key Binding Request is valid and verifiable using the provided ``jwk``.
  3. The ``nonce`` value has been generated by the Relying Party Backend and not previously used.
  4. The Relying Party Instance has valid Cryptographic Hardware Keys registered.
  5. The ``client_data`` can be reconstructed using ``nonce`` and ``cnf`` (representing ``key_pub``). The ``hardware_signature`` parameter value is then validated using the registered Cryptographic Hardware Key's public key associated with the Relying Party Instance.
  6. The ``key_attestation`` can be validated according to the device manufacturer's guidelines. The specific checks performed by the Relying Party Backend are detailed in the operating system manufacturer's documentation.
  7. The device in use is free of known security flaws and meets the minimum security requirements defined by the Relying Party.
  8. The URL in the ``iss`` parameter matches the Relying Party's URL identifier.

> **Step 13:** If the checks are successful, the :ref:`Relying Party Key Binding Endpoint` responds with a confirmation of success (:ref:`Relying Party Key Binding Response`).

Reviewer: ```suggestion
**Step 13:** If the checks are successful, the Relying Party Backend responds with a confirmation of success (:ref:`Relying Party Key Binding Response`).
```

**Step 14:** The Mobile Relying Party Instance generates a Certificate Signing Request (CSR, ``csr``) using ``key_pub`` and ``key_priv``.

**Step 15:** The Mobile Relying Party Instance sends the CSR to the :ref:`Relying Party Access Certificate Endpoint` of the Relying Party Backend, as part of the :ref:`Relying Party Access Certificate Request`.

> **Steps 16-17:** The :ref:`Relying Party Access Certificate Endpoint` checks that the public key in the CSR corresponds to a Relying Party Instance that has been previously validated, i.e., that it matches the one bound to the Cryptographic Hardware Keys through ``hardware_signature``. If this check is successful, the Relying Party Backend sends the CSR to the Relying Party Instance Access Certificate Authority.

Reviewer: ```suggestion
**Steps 16-17:** The Relying Party Backend checks that the public key in the CSR corresponds to a Relying Party Instance that has been previously validated, i.e., that it matches the one bound to the Cryptographic Hardware Keys through ``hardware_signature``. If this check is successful, the Relying Party Backend sends the CSR to the Relying Party Instance Access Certificate Authority.
```

**Steps 18-19:** The Relying Party Instance Access Certificate Authority signs the CSR, obtaining a valid Access Certificate (``access_certificate``) which it sends back to the Relying Party Backend.

> **Steps 20-21:** The :ref:`Relying Party Access Certificate Endpoint` sends the Access Certificate (as part of the :ref:`Relying Party Access Certificate Response`) to the Mobile Relying Party Instance, which stores it for future authentication towards Wallet Instances.

Reviewer: ```suggestion
**Steps 20-21:** The Relying Party Backend sends the Access Certificate (as part of the :ref:`Relying Party Access Certificate Response`) to the Mobile Relying Party Instance, which stores it for future authentication towards Wallet Instances.
```
